### ISSUE TITLE
perl: dependency on gcc and others

### ISSUE THREAD

**Issue by vcunat** [NixOS/nixpkgs#10244]
Does anyone know the use of ``${perl}/lib/perl5/${version}/${system}-thread-multi/Config*``? There's various information gathered. Some make (or made) the build non-deterministic and some make it depend on gcc...

**Comment by fictitious** [comment]
With regard to gcc - if I'm not mistaken, it keeps path to c compiler and all the flags that were used when perl was compiled, so that perl modules written in C and installed later would be compiled by the same compiler with the same flags. 

Brief description of Config can be found here http://search.cpan.org/~shay/perl-5.20.3/configpm , 
list of all confg variables is documented here: http://perl5.git.perl.org/perl.git/blob/HEAD:/Porting/Glossary 

TL;DR it's there so that any random perl code can know exactly how this particular perl installation was configured.

**Comment by vcunat** [comment]
Thanks for the links. I'll try to clear some of the information carefully. It shouldn't happen too much to the scripts when include and library paths are lost from the config.

**Comment by vcunat** [comment]
Well, since b5636e9 perl only has libc and coreutils in the closure. It seems we can do no better.

**Comment by vcunat** [comment]
... at least no better with closure reductions. Lets consider the non-determinism consequences a separate issue.
